XRTK-Core/Packages/com.xrtk.core/Inspectors/Profiles/InputSystem/Controllers/BaseMixedRealityControllerDataProviderProfileInspector.cs

```diff
@@ -168,17 +171,86 @@ public override void OnInspectorGUI()
         {
             RenderHeader("This profile defines all of the controllers and their mappings to use. Additional platform settings can also be available as well.");
 
+            showMappingProfiles = EditorGUILayoutExtensions.FoldoutWithBoldLabel(showMappingProfiles, new GUIContent("Controller Mapping Profiles"), true);
+            if (showMappingProfiles)
+            {
+                EditorGUILayout.Space();
+                selectedMappingsViewModeTab = GUILayout.Toolbar(selectedMappingsViewModeTab, new string[] { "Simple", "Advanced" });
+                switch (selectedMappingsViewModeTab)
+                {
+                    case 0:
+                        DrawSimpleControllerMappingProfilesView();
+                        break;
+                    case 1:
+                        DrawAdvancedControllerMappingProfilesView();
+                        break;
+                }
+            }
+
+            EditorGUILayout.Space();
+        }
+
+        private void DrawAdvancedControllerMappingProfilesView()
+        {
             serializedObject.Update();
+            mappingProfileList.DoLayoutList();
+            serializedObject.ApplyModifiedProperties();
+        }
 
-            showMappingProfiles = EditorGUILayout.Foldout(showMappingProfiles, new GUIContent("Controller Mapping Profiles"), true);
+        private void DrawSimpleControllerMappingProfilesView()
+        {
+            if (controllerButtonStyle == null)
```

Comment:
can't we do this in OnEnable?

Reply:
Has to be in the OnGUI loop since you can change the selection any time.

Reply:
Ah sorry now I get what you meant. You're talking about the button style. Unfortunately it has to be in the OnGUI loop. If If put it in OnEnable Unity throws an error to put it in OnGUI. Some dark GUI Styles magic going on there.